backend/inspirehep/records/marshmallow/literature/ui.py

```diff
@@ -187,6 +189,26 @@ def add_ads_links_for_arxiv_papers(self, data):
 
         return data
 
+    @staticmethod
+    def is_document_hidden(document):
+        return document.get("hidden", False)
+
+    @staticmethod
+    def is_document_local(document):
+        return current_s3_instance.is_public_url(document["url"])
```

Comment:
not sure about it, now we're using ``/files/...` will it work? Also I think it's unecessary for `get_citation_pdf_urls` since `documents` with `hidden:false` are always processed by us right?

Reply:
Yes we are using `/files/` and `is_public_url` knows about this and checks if url is ours and is served with public url. 
Files with hidden:False are always processed by us, but if a record had an error on downloading file then this data won't be updated. So there is a chance that there will be a public file with url which is not local.